### PR TITLE
Update: Updated UI for transfering a project namespace

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?

Moves transferring a project to a different namespace into it's own card container and introducing an "Advanced settings" section within the edit page.

## Screenshots or screen recordings

Updated project settings panel:

Updated confirmation dialog:

## How to set up and validate locally

1.  Starting with a fresh development database, login with user1
2.  Go to the settings page for [**Streptococcus / Streptococcus pyogenes / Outbreak 2022**](http://localhost:3000/streptococcus/streptococcus-pyogenes/outbreak-2022/-/edit)
3.  You will see the updated page with a new "Advanced settings" > "Transfer Project"
4.  The "Transfer project" button should be disabled
5.  From the "Select a new namespace" select input, select `S`treptococcus / Streptococcus agalactiae"
6.  The "Transfer project" button should become enabled, click on it.
7.  A confirm dialog should be displayed where you need to type the project name, enter "Outbreak 2022" and click "Confirm"
8.  The project cannot be transferred to the namespace and an error message should be displayed under the select input.
9.  From the "Select a new namespace" select, select "user1@email.com", and click the "Transfer project" button and repeat the confirmation.
10.  This transfer works and the user is redierected back to the main project page.  If you look in the breadcrumbs, you will see that the namespace has been updated (also in the url).

### CODE REVIEW

File: app/controllers/projects/transfer_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        end
+      else
+        @error = @project.errors.messages.values.flatten.first
+        render :edit, status: :unprocessable_entity
+      end
+    end
+
+    private
+
+    def project_params
```
Comment (ericenns) [inline app/controllers/projects/transfer_controller.rb:23]: **nitpick (blocking):** Please remove this, this looks to be unintentionally copied over.

Comment (ericenns) [inline app/controllers/projects/transfer_controller.rb:23]: **nitpick (blocking):** Please remove this, this looks to be unintentionally copied over.

File: app/controllers/projects/transfer_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      params.require(:project)
+            .permit(project_params_attributes)
+    end
+
+    def new_namespace
+      id = params.require(:new_namespace_id)
+      Namespace.find_by(id:)
+    end
+
+    def namespace_attributes
```
Comment (ericenns) [inline app/controllers/projects/transfer_controller.rb:33]: **nitpick (blocking):** Please remove this, this looks to be unintentionally copied over.

File: app/controllers/projects/transfer_controller.rb (hunk 1 above)
Comment (ericenns) [inline app/controllers/projects/transfer_controller.rb:23]: **nitpick (blocking):** Please remove this, this looks to be unintentionally copied over.

File: app/controllers/projects/transfer_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      %i[
+        name
+        path
+        description
+        parent_id
+      ]
+    end
+
+    def project_params_attributes
```
Comment (ericenns) [inline app/controllers/projects/transfer_controller.rb:42]: **nitpick (blocking):** Please remove this, this looks to be unintentionally copied over.

**Review by ericenns — CHANGES_REQUESTED** [review]
Looking good except for some extra methods. I still need to run the code, but so far looking great.

File: app/controllers/projects/transfer_controller.rb (hunk 2 above)
Comment (joshsadam) [reply app/controllers/projects/transfer_controller.rb]: Removed in fa24b29

File: app/controllers/projects/transfer_controller.rb (hunk 1 above)
Comment (joshsadam) [reply app/controllers/projects/transfer_controller.rb]: Removed in fa24b29

File: app/controllers/projects/transfer_controller.rb (hunk 3 above)
Comment (joshsadam) [reply app/controllers/projects/transfer_controller.rb]: Removed in fa24b29

File: config/tailwind.config.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ module.exports = {
     "./node_modules/flowbite/**/*.js",
     "./test/components/previews/**/*",
     "./test/components/viral/system/**/*",
+    "./config/locales/en.yml",
```
Comment (ericenns) [inline config/tailwind.config.js:11]: tailwind needs translations?

Comment (ericenns) [inline config/tailwind.config.js:11]: Can you remove this line, I don't think tailwind needs access to the translations

File: test/controllers/projects/transfer_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    test 'should not create a new transfer with wrong parameters' do
+      post namespace_project_transfer_index_path(@old_namespace, @project),
+           params: { new_namespace_id: 0 }, as: :turbo_stream
+
+      assert_response :unprocessable_entity
+    end
+
+    test 'should not transfer a project to unowned namespace' do
+      post namespace_project_transfer_index_path(@old_namespace, @project),
```
Comment (ericenns) [inline test/controllers/projects/transfer_controller_test.rb:29]: Shouldn't this also use `as: :turbo_stream`

File: app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -64,18 +64,6 @@ def activity
     # No necessary code here
   end
 
-  def transfer
```
Comment (ericenns) [inline app/controllers/projects_controller.rb:67]: Instead of removing this and moving the method into a new transfer controller can you move it back here. I did some digging and you are not using the `new` endpoint of the transfer controller you added and only the `create` as such I think it would be simpler to just keep this method and make it match the other method you had added.

File: config/routes/project.rb [hunk 7]
```diff
@@ -15,7 +15,7 @@
     scope '-' do
       get :activity
       get :edit
-      post :transfer
+      resources :transfer, only: %i[create new]
```
Comment (ericenns) [inline config/routes/project.rb:18]: As per comment above, let's undo this change and just use the `transfer` endpoint instead.

File: test/controllers/projects_controller_test.rb [hunk 8]
```diff
@@ -118,44 +118,6 @@ class ProjectsControllerTest < ActionDispatch::IntegrationTest # rubocop:disable
     assert_response :unauthorized
   end
 
-  test 'should fail to update a project with wrong params' do
```
Comment (ericenns) [inline test/controllers/projects_controller_test.rb:121]: Can we keep this method for now, and then update/remove it when you actually change the process to rename a project or change it's path.

**Review by ericenns — CHANGES_REQUESTED** [review]
I found some issues with the new path you added in the transfer controller, instead I think we can move the functionality back into the projects controller. See comments below.

File: config/tailwind.config.js (hunk 4 above)
Comment (joshsadam) [reply config/tailwind.config.js:11]: We actually do need it in there.  It allows for doing things like on line 408 and 411.  Without having tailwind scan these we cannot have this mark up.  Doing it another way would be extremely complicated.

File: test/controllers/projects/transfer_controller_test.rb (hunk 5 above)
Comment (joshsadam) [reply test/controllers/projects/transfer_controller_test.rb:29]: Makes sense. Updated in 6179663

File: config/tailwind.config.js (hunk 4 above)
Comment (joshsadam) [reply config/tailwind.config.js:11]: Just thought of something, I will try tomorrow morning

Comment (joshsadam) [reply config/tailwind.config.js]: Your right, better solution is just to create a custom css class.  See 58bd686

File: config/routes/project.rb (hunk 7 above)
Comment (joshsadam) [reply config/routes/project.rb]: Reverted in 2703d62

File: app/controllers/projects_controller.rb (hunk 6 above)
Comment (joshsadam) [reply app/controllers/projects_controller.rb:67]: Sounds good, updated in 2703d62

File: app/views/projects/transfer/new.html.erb
Comment (ericenns) [inline app/views/projects/transfer/new.html.erb:1]: **nitpick (blocking):** Can this empty file be removed

File: app/views/projects/transfer/_edit_transfer.html.erb
Comment (ericenns) [inline app/views/projects/transfer/_edit_transfer.html.erb:1]: **nitpick (blocking):** Can we move all the files in the `transfer` folder back into the `projects` folder. This file can also be renamed to `_transfer.html.erb`

File: app/views/projects/transfer/_form.html.erb
Comment (ericenns) [inline app/views/projects/transfer/_form.html.erb:1]: **nitpick (blocking):** Can we move all the files in the `transfer` folder back into the `projects` folder. This file can also be renamed to `_transfer_form.html.erb`

File: app/views/projects/transfer/edit.turbo_stream.erb
Comment (ericenns) [inline app/views/projects/transfer/edit.turbo_stream.erb:1]: **nitpick (blocking):** Can we move all the files in the `transfer` folder back into the `projects` folder. This file can also be renamed to `transfer.turbo_stream.erb`

File: app/views/projects/transfer/new.html.erb
Comment (joshsadam) [reply app/views/projects/transfer/new.html.erb:1]: Updated in 1b24c1c

File: app/views/projects/transfer/_edit_transfer.html.erb
Comment (joshsadam) [reply app/views/projects/transfer/_edit_transfer.html.erb:1]: Updated in 1b24c1c

File: app/views/projects/transfer/_form.html.erb
Comment (joshsadam) [reply app/views/projects/transfer/_form.html.erb:1]: Updated in 1b24c1c

File: app/views/projects/transfer/edit.turbo_stream.erb
Comment (joshsadam) [reply app/views/projects/transfer/edit.turbo_stream.erb:1]: Updated in 1b24c1c

**Review by ericenns — DISMISSED** [review]

File: app/views/projects/edit.html.erb [hunk 9]
```diff
@@ -1,47 +1,45 @@
...
+                                     required: true,
+                                     autofocus: true,
+                                     placeholder: t(:"projects.new.placeholder") %>
+            </div>
+            <div class="form-field">
+              <%= builder.label :description %>
+              <%= builder.text_area :description,
+                                    {
+                                      :class => "form-control",
+                                      "aria-label" => "Project description, not required"
```
Comment (ericenns) [inline app/views/projects/edit.html.erb:25]: **nitpick (blocking):** Could you add a translation here please

**Review by ericenns — CHANGES_REQUESTED** [review]
This is looking great! I found one spot where a translation is missing otherwise this is working as expected

File: app/views/projects/edit.html.erb (hunk 9 above)
Comment (joshsadam) [reply app/views/projects/edit.html.erb]: Updated in b6f1d20

File: config/locales/en.yml [hunk 10]
```diff
@@ -398,7 +398,33 @@ en:
       title: Settings
       general:
         title: General
+        heading: Project name and description
         submit: Update project
+      advanced:
+        title: Advanced settings
+        path:
+          title: Change path
+          description: A project's path is a unique identifier where it can be accessed in the URL
```
Comment (deepsidhu85) [inline config/locales/en.yml:407]: Can you update this text to read `A project's path is a unique identifier that can be accessed via the URL`

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good to me. Just the one comment below

File: config/locales/en.yml (hunk 10 above)
Comment (joshsadam) [reply config/locales/en.yml]: Sure, updated in 52534af Thanks

Comment (deepsidhu85) [reply config/locales/en.yml]: Awesome thanks!

**Review by deepsidhu85 — APPROVED** [review]

**Review by ericenns — APPROVED** [review]
